nixos/tests/appliance-gpt-image.nix

```diff
@@ -0,0 +1,114 @@
+# Tests building and running a GUID Partition Table (GPT) appliance image.
+# "Appliance" here means that the image does not contain the normal NixOS
+# infrastructure of a system profile and cannot be re-built via
+# `nixos-rebuild`.
+
+{ lib, ... }:
+
+let
+  rootParitionlabel = "root";
+
+  bootLoaderConfigPath = "/loader/entries/nixos.conf";
+  kernelPath = "/EFI/nixos/kernel.efi";
+  initrdPath = "/EFI/nixos/initrd.efi";
+in
+{
+  name = "appliance-gpt-image";
+
+  meta.maintainers = with lib.maintainers; [ nikstur ];
+
+  nodes.machine = { config, lib, pkgs, ... }: {
+
+    imports = [ ../modules/image/gpt.nix ];
+
+    virtualisation.directBoot.enable = false;
+    virtualisation.mountHostNixStore = false;
+    virtualisation.useEFIBoot = true;
+
+    # Disable boot loaders because we install one "manually".
+    boot.loader.grub.enable = false;
+
+    virtualisation.fileSystems = lib.mkForce {
+      "/" = {
+        device = "/dev/disk/by-partlabel/${rootParitionlabel}";
```

Comment:
```suggestion
        device = "/dev/disk/by-partlabel/${rootPartitionlabel}";
```

Reply:
Don't you just need to override `virtualisation.rootDevice` here?

Reply:
I believe this would make this less readable and more complex. The right way to use this image is via the changes from https://github.com/NixOS/nixpkgs/pull/238847

Reply:
Then, I disagree, I feel like you should do `filesystems."/" = mkImageMediaOverride { ... }` and ensure that this works out of the box in virtualization because `mkImageMediaOverride` should be of higher priority.
Otherwise, we can consider a higher priority override.

Then, your changes of #238847 can become simpler I believe.

Reply:
This is the same issue as the netboot media and the ISO media.